### PR TITLE
Add support for filepath repository rules

### PR DESCRIPTION
**Pull request by grahamhar**
Fixes: #3231.

This is an attempt to add support for file_path_restriction in repository rules. It will implement #3231

### CODE REVIEW

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @grahamhar!
Just a couple minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos_rules.go
Comment (grahamhar) [reply github/repos_rules.go]: done

**Review by gmlewis — APPROVED** [review]
Thank you, @grahamhar!
In the future, please don't force-push in this repo as it makes code reviews more challenging. For this PR it was no big deal, but when the PRs get bigger, it becomes an issue.  We always squash-and-merge PRs anyway, so there is no need for a force-push. Thanks.

LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Comment by grahamhar** [comment]
@gmlewis sorry to trouble you but when is the next release due?

**Comment by gmlewis** [comment]
> @gmlewis sorry to trouble you but when is the next release due?

No troubles.  It looks like we are due for a roughly-monthly release so I will try to get this done by tonight (EDT).

**Comment by gmlewis** [comment]
> @gmlewis sorry to trouble you but when is the next release due?

Release completed: https://github.com/google/go-github/releases/tag/v64.0.0

**Pull request merged**
